cocos/core/assets/effect-asset.ts

```diff
@@ -260,7 +260,7 @@ export class EffectAsset extends Asset {
     public onLoaded () {
         programLib.register(this);
         EffectAsset.register(this);
-        if (!EDITOR || legacyCC.GAME_VIEW) { legacyCC.game.once(legacyCC.Game.EVENT_ENGINE_INITED, this._precompile, this); }
+        if (!EDITOR || legacyCC.GAME_VIEW) { legacyCC.game.once(legacyCC.Game.EVENT_RENDERER_INITED, this._precompile, this); }
```

Comment:
Why change EVENT_ENGINE_INITED to EVENT_RENDERER_INITED.
Will it break the compatibility?

Reply:
The previous usage was wrong, but as the previous built-in effect was not generated by the generic process, some of the data was missing, and it happened to work under the previous usage. In the normal flow, effect-assets need to wait for the pipeline and renderer initialization to complete before they can be compiled, so here they are adjusted to RENDERER_INITED